docs/tutorials/07_gradient_descent.ipynb

```diff
@@ -784,47 +792,50 @@
     "            yield self.inputs[start:end], self.labels[start:end]"
    ]
   },
-  {
-   "cell_type": "code",
-   "execution_count": 29,
-   "id": "b416c56b",
-   "metadata": {},
-   "outputs": [],
-   "source": [
-    "batch_size = 4\n",
-    "\n",
-    "dataloader = Dataset(inputs, labels)\n",
-    "dataloader = dataloader.shuffle(seed=1).batch(batch_size)"
-   ]
-  },
   {
    "cell_type": "markdown",
-   "id": "8786a19b",
+   "id": "d7455f4f",
    "metadata": {},
    "source": [
     "### Training loop"
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": null,
-   "id": "712276fe",
+   "execution_count": 29,
+   "id": "723e6f5f",
    "metadata": {},
    "outputs": [
     {
      "name": "stdout",
      "output_type": "stream",
      "text": [
-      "epoch 0, loss 25.09776566535514\n"
+      "epoch 0, loss 25.097769115066576\n",
+      "epoch 1, loss 21.143421957090222\n",
+      "epoch 2, loss 15.152694619943718\n",
+      "epoch 3, loss 8.959925803536182\n",
+      "epoch 4, loss 6.721850849093422\n",
+      "epoch 5, loss 6.207057397100044\n",
+      "epoch 6, loss 6.136331377206082\n",
+      "epoch 7, loss 6.102978172393111\n",
+      "epoch 8, loss 6.051937035793608\n",
+      "epoch 9, loss 6.105838843977726\n"
      ]
     }
    ],
    "source": [
+    "batch_size = 4\n",
+    "\n",
     "for epoch in range(10):\n",
     "    epoch_loss = 0.0\n",
+    "    \n",
+    "    # Our simple dummy dataloader must be re-initialized at every epoch.\n",
+    "    dataloader = Dataset(inputs, labels)\n",
+    "    dataloader = dataloader.shuffle(seed=epoch).batch(batch_size)\n",
+    "\n",
     "    for batch_ind, batch in enumerate(dataloader):\n",
-    "        current_batch = batch[0].numpy()\n",
```

Comment:
Could also be changed to  https://github.com/jaxleyverse/jaxley/blob/0df50b6aa7338ffe16943afbac410f2616b2e4bb/docs/tutorials/07_gradient_descent.ipynb#L826